src/schoology-extractor/schoology_extractor/api/request_client.py

```diff
@@ -34,19 +45,14 @@ class RequestClient:
     ----------
     oauth : OAuth1Session
         The two-legged authenticated OAuth1 session.
-    '''
+    """
 
-    def __init__(
-        self, schoology_key: str, schoology_secret: str, base_url: str = DEFAULT_URL
-    ):
-        assert isinstance(schoology_key, str), "Argument `schoology_key` should be of type `str`."
-        assert isinstance(schoology_secret, str), "Argument `schoology_secret` should be of type `str`."
-        assert isinstance(base_url, str), "Argument `base_url` should be of type `str`."
+    schoology_key: str
+    schoology_secret: str
+    base_url: str = DEFAULT_URL
 
-        self.oauth = OAuth1Session(schoology_key, schoology_secret)
-        self.base_url = base_url
-        self.consumer_key = schoology_key
-        self.consumer_secret = schoology_secret
+    def __post_init__(self):
```

Comment:
I'd move the asserts for the properties to this method... What I don't like about having them in the _request_header prop, is that if later we need to add another property or method, is not clear what values we need to validate or where, while if we have them here, it would be like a central point to evaluate all the required properties for the class that should be passed in the constructor.

Reply:
Makes sense to me